### ISSUE TITLE
Feature Request: upload towards google photos

### ISSUE THREAD

**Issue by MijnSpam** [photoprism/photoprism#4507]
**Describe what problem this solves and why this would be valuable to many users**

I would like to be able to upload / sync towards google photos. Ideally only a marked folder / only newly added items.

**Describe the solution you'd like**

It seems the only focus for now is to download from google photos but from a backup perspective ( 3 2 1) my phone is primary device, PhotoPrism is primary backup and GP would be secondary backup. 
In this setup I only need to upload the photo's to PhotoPrism and PP would then take care of the secondary backup towards google photos saving time and data plan for my mobile. 

**Describe alternatives you've considered**

The only option for now is doing 2 uploads (1x photoprism, 1x google photos) this is not desirable.
Looked for a webdav option in google photos but could find any.

**Comment by srett** [comment]
I don't see why this should be integrated into PhotoPrism specifically. This seems out of scope, as then you might ask why not add support for uploading to service X, or via protocol Y.

There seem to be a bunch of tools around on github that allow uploading to google photos. The only question would be whether these tools allow incremental uploads, i.e. check which files already exist remotely. If so, set up a cronjob or similar, to trigger an upload once a day.

That being said, I'm not sure Google Photos is a good backup solution in general. Yes it's free, but AFAIK, if you download your files again, some EXIF Data gets stripped, and videos might get transcoded to a lower bitrate.

**Comment by MijnSpam** [comment]
I understand what you say but as PP is running in a podman container getting a script in there is a bit like working with a Pandora's box for me ( ok still not a PP problem).
Having an import script function would be nice as well but that is not this feature requests.

And like I said getting an export to Google photos is a request maybe not granted but if you don't ask you don't get an answer

**Comment by srett** [comment]
> I understand what you say but as PP is running in a podman container getting a script in there is a bit like working with a Pandora's box for me ( ok still not a PP problem).

But surely you're mounting an external directory into the container for persistent storage of your photos? Can't you just set up the backup script outside the container? Either on the host machine directly, or add another container that also bind-mounts the originals directory and is responsible just for the backups.

**Comment by MijnSpam** [comment]
It's mounted through OMV so basically I didn't have to do any real setup. And a share photos is shared with photoprism. 
Having a second look at the "plugin" I gues the optional application data could be use. But then it's still a big questionmark how to do a export for everything newly uploaded to PP

**Comment by rayzorben** [comment]
I agree. My scenario is I don't use GP for backup, but I have multiple Google TVs that show specific GP albums as screensavers. I want to create an album in PhotoPrism, sync it to GP to display on my TVs so only selected photos are shown.

**Comment by strugee** [comment]
A related usecase: I want to be able to send a PhotoPrism link to friends using Google Photos, and allow them to easily add the album to their library.

Happy to file a separate issue for that usecase since it's unclear to me how much it would overlap with the feature request here.

**Comment by lastzero** [comment]
Could you then also do some research to see if that's technically possible? If so, could you suggest a specific implementation and list any constraints? Thank you! 🤗
